### PR TITLE
Implement new gidRanges attribute to the existing code

### PR DESCRIPTION
**Pull request by stavamichal**
- all methods in ModulesUtils which works with gid was reworked to use
   new attribute gidRanges intstead old attributes minGID and maxGID
 - fix javadocs where minGID and maxGID were mentioned
 - remove unused methods from ModulesUtilsBlImpl
 - add both new attributes to the initialization of attributes
 - add info about deprycated classes of old modules minGID and maxGID
   (entityless attributes and also virtual attributes for facility)
 - replace old depepndencies where needed
 - fix typo in name of attribute in module namespace-GIDRanges
 - fix tests where old behavior was used
 - remove not used test class

### CODE REVIEW

**Comment by stavamichal** [comment]
This code can be merged only if new attributes with gid ranges will be set on all instances with GID namespaces.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 1]
```diff
@@ -6091,6 +6093,26 @@ protected void initialize() throws InternalErrorException {
...
+		//set attribute rights (with dummy id of attribute - not known yet)
+		rights = new ArrayList<>();
+		rights.add(new AttributeRights(-1, Role.SELF, Collections.singletonList(ActionType.READ)));
+		rights.add(new AttributeRights(-1, Role.VOADMIN, Collections.singletonList(ActionType.READ)));
+		rights.add(new AttributeRights(-1, Role.GROUPADMIN, Collections.singletonList(ActionType.READ)));
+		rights.add(new AttributeRights(-1, Role.FACILITYADMIN, Collections.singletonList(ActionType.READ)));
+		attributes.put(attr, rights);
+
+		//urn_perun_entityless_attribute_def_def_namespace_GIDRanges
+		attr = new AttributeDefinition( (new urn_perun_entityless_attribute_def_def_namespace_GIDRanges()).getAttributeDefinition() );
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: I'm not sure, that user or vo/group admin need to read gidRanges entityless attribute. This is usually managed only by perun admin anyway.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 2]
```diff
@@ -6091,6 +6093,26 @@ protected void initialize() throws InternalErrorException {
 		rights.add(new AttributeRights(-1, Role.FACILITYADMIN, Collections.singletonList(ActionType.READ)));
 		attributes.put(attr, rights);
 
+		//urn_perun_facility_attribute_def_virt_GIDRanges
+		attr = new AttributeDefinition( (new urn_perun_facility_attribute_def_virt_GIDRanges()).getAttributeDefinition() );
+		//set attribute rights (with dummy id of attribute - not known yet)
+		rights = new ArrayList<>();
+		rights.add(new AttributeRights(-1, Role.SELF, Collections.singletonList(ActionType.READ)));
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: Not sure, that user need to read this, it's probably interesting for vo/group manager for support use-cases.

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java [hunk 3]
```diff
@@ -40,6 +40,8 @@
 import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
 import cz.metacentrum.perun.core.bl.ModulesUtilsBl;
 import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import sun.awt.image.ImageWatched;
```
Comment (zlamalp) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java]: Probably bad autoimport by IDEA and it shouldn't be here.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java (hunk 2 above)
Comment (stavamichal) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: We don't want to create any restriction on rights to read value of this attribute.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java (hunk 1 above)
Comment (stavamichal) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: We don't want to create any restriction on rights to read value of this attribute.

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java (hunk 3 above)
Comment (stavamichal) [reply perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java]: Removed.

**Pull request merged**